### ISSUE TITLE
Authorization fails only when Spotify app is installed

### ISSUE THREAD

**Issue by Davidorenlane** [spotify/ios-sdk#183]
I'm using the Authorization code flow, but when I call initiateSession and am redirected to the Spotify app, I see the "authorizing ..." pop-up but then when it finishes I get an "X" rather than the usual checkmark and am redirected back to my app. No sessionManager delegate function (didFailWith, for example) is called and my app just hangs. 

Of note: 
- if I uninstall Spotify and use the web view authorization to authorize, rather than SFAuthenticationSession, I authorize just fine.
- I cannot force this by setting `sessionManager.alwaysShowAuthorizationDialog = true`, it still just flips to Spotify and starts "authorizing ..." (and then fails) regardless
- running the code in the XCode simulator works.
- as of today where latest iOS is 13.2, my issue is occurring on devices running both iOS 13.1.3 and 12.1.3
- I recently lowered the deployment target for my app to iOS 12.0, around then is when I started seeing issues

The only workaround that I have found so far is to remove Spotify from LSApplicationsQueriesScheme in my info.plist. This forces my app to use the web view authorization flow. Obviously this is slower and much worse UX than an automatic authorization using SFAuthentificationSession. 

Does anyone know why this might be happening? I also don't know how to get logs from the call to sessionManager.initiateSession so any help with that would also be greatly appreciated

**Comment by rasmuskarlsten** [comment]
@Davidorenlane Did you find a better solution to this problem? I am having the exact same problem right now. Works perfectly fine in XCode simulator but not on my iPhone.

**Comment by Davidorenlane** [comment]
@rasmuskarlsten Unfortunately, no. I hope the workaround is good enough though :/ After the first time a user manually types in their credentials, their credentials will save and the UX will be pretty much the same as with the automatic authorization. Try it out if you haven't already.

**Comment by bestoml** [comment]
You forget one thing in AppDelegate.swift. Add the code to process the response from spotify. After this process , delegate of sessionManager will be called.
func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool {
        rootViewController.sessionManager.application(app, open: url, options: options)
        return true
}
But after ios 13, it wouldn't call thie func above.You need to do as follow.
1. Add openURLContexts func in SceneDelegate file to process the same thing of openURL in appdelegate.
- (void)scene:(UIScene *)scene openURLContexts:(NSSet<UIOpenURLContext *> *)URLContexts {
UIOpenURLContext *ctx = [URLContexts allObjects][0];
ViewController *rootVC = [UIApplication sharedApplication].keyWindow.rootViewController;
[rootVC.sessionManager application:[UIApplication sharedApplication] openURL:ctx.URL options:ctx.options];
}
2. remove UISceneConfigurations section from info.plist. Then it will callback to openurl.

**Comment by Davidorenlane** [comment]
@rasmuskarlsten  I solved this problem and wanted to share my solution:

The Bundle ID I had listed in my Spotify Dashboard settings was antiquated and no longer was the same as the bundle ID in my XCode project. In my Info.plist, I was using the CFBundleURLName of the XCode project, not what was in the dashboard, so I added my XCode bundle ID to my Spotify project dashboard, and now it is working again. 

Let me know if this works for you!
